src/de/neuland/firefly/HybrisAdapter.java

```diff
@@ -53,12 +58,33 @@ private void doInitialize(boolean update) throws Exception {
             request.addParameter("clearHMC", "true");
 
             JspContext jspContext = new JspContext(jspWriter, request, new MockHttpServletResponse());
-            Initialization.doInitialize(jspContext);
+
+            try {
+                InitializationLockHandler handler = new InitializationLockHandler(new DefaultInitLockDao());
+                String operationName = update ? "System update" : "HMC Reset";
+
+                handler.performLocked(Registry.getCurrentTenant(),
+                                      createUpdateCallable(jspContext),
+                                      operationName);
+            } catch (NoSuchMethodException e) {
+                Initialization.doInitialize(jspContext);
+            }
         } finally {
             LOG.debug(de.hybris.platform.util.Utilities.filterOutHTMLTags(jspWriter.getString()));
         }
     }
 
+    private Callable<Boolean> createUpdateCallable(final JspContext jspContext) {
```

Comment:
consider renaming to `createInitializeCallable`

Reply:
done